packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "../fixtures/optimistic-assertor/OptimisticAssertorFixture.sol";
+import "../fixtures/common/TestAddress.sol";
+
+contract OptimisticAsserterLifecycle is Test {
+    OptimisticAssertor optimisticAssertor;
+    TestnetERC20 defaultCurrency;
+    Timer timer;
+    string claimAssertion = 'q:"The sky is blue"';
+
+    function setUp() public {
+        OptimisticAssertorFixture.OptimisticAsserterContracts memory oaContracts =
+            new OptimisticAssertorFixture().setUp();
+        optimisticAssertor = oaContracts.optimisticAssertor;
+        defaultCurrency = oaContracts.defaultCurrency;
+        timer = oaContracts.timer;
+    }
+
+    function testAssertionWithNoDispute() public {
+        vm.startPrank(TestAddress.account1);
+        defaultCurrency.allocateTo(TestAddress.account1, optimisticAssertor.defaultBond());
+        assert(defaultCurrency.balanceOf(TestAddress.account1) >= optimisticAssertor.defaultBond());
+        defaultCurrency.approve(address(optimisticAssertor), optimisticAssertor.defaultBond());
+
+        bytes32 assertionId = optimisticAssertor.assertTruth(bytes(claimAssertion));
+
+        // Move time forward to the end of the liveness period.
+        timer.setCurrentTime(timer.getCurrentTime() + optimisticAssertor.defaultLiveness());
+
+        // The assertion should be true.
+        assert(optimisticAssertor.settleAndGetAssertion(assertionId));
```

Comment:
as we are using `assertTruth` that currently passes `address(0)` as `proposer` I wonder why this test did not fail? internal `settleAssertion` should have reverted due to non-existing `proposer` address!

Reply:
nevermind, my mistake as internal `assertTruthFor` by default passes msg.sender